### PR TITLE
docs: added prereqs, docker instructions, and dbmate instructions

### PR DESCRIPTION
**Pull request by tucker-weed**
Description

1. Added install prerequisites
2. Added instructions to build and start the dockerized services
3. Added instructions for creating, running, and rolling back migrations with dbmate

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -9,13 +9,34 @@ Using CRDTs to build a collaborative text editor, is mostly a solved problem. Tu
 
 [Repo](https://github.com/chewnoill/collaborative-editor)
 
+## Prerequisites
+
+1. node
+2. docker
+3. postgres
```
Comment (chewnoill) [inline readme.md:16]: postgres is installed by docker, is not a direct dependency.

Comment (tucker-weed) [reply readme.md:16]: Would just listing docker and dbmate be the way to go @chewnoill ?

Comment (chewnoill) [reply readme.md:16]: yeah, we only need to list things the developer has to have installed get get started

File: readme.md [hunk 2]
```diff
@@ -9,13 +9,34 @@ Using CRDTs to build a collaborative text editor, is mostly a solved problem. Tu
...
+### Install dependencies:
 ```shell
 yarn
 ```
 
+### Docker setup:
+
+build
+```shell
+docker build && docker-compose build
```
Comment (chewnoill) [inline readme.md:30]: `docker-compose build` and `docker build` are redundant, you don't need to do both.

File: readme.md [hunk 3]
```diff
@@ -26,6 +47,19 @@ start storybook
 yarn storybook
 ```
 
+start postgres
+```shell
+postgres -D <data_directory>
+```
```
Comment (chewnoill) [inline readme.md:53]: we don't need this at all

File: readme.md [hunk 4]
```diff
@@ -26,6 +47,19 @@ start storybook
...
+
+## Running database migrations using dbmate
+
+Creating migrations: https://github.com/amacneil/dbmate#creating-migrations
+
+Running migrations: https://github.com/amacneil/dbmate#running-migrations
+
+Rolling back migrations: https://github.com/amacneil/dbmate#rolling-back-migrations
```
Comment (chewnoill) [inline readme.md:61]: Its okay to link to external sources for more details, but this readme should include the commands needed to create and run migrations. Especially if we're running it in docker.

File: readme.md (hunk 1 above)
Comment (chewnoill) [reply readme.md:16]: actually I needed to `brew install postgresql` to get dbmate working. I needs `pg_dump`

**Review by chewnoill — APPROVED** [review]

**Pull request merged**